Engine/Results/BacktestingResultHandler.cs

```diff
@@ -772,14 +775,43 @@ public virtual void ProcessSynchronousEvents(bool forceProcess = false)
             if (Algorithm == null) return;
 
             var time = Algorithm.UtcTime;
+            var shouldSample = _previousProcessTime.Date != time.Date;
+            var currentPortfolioValue = Algorithm.Portfolio.TotalPortfolioValue;
+
+            if (shouldSample)
+            {
+                // Sample all of the metrics daily once we're certain that all feeds
+                // and prices have been setup properly. In the past, we sampled at the beginning of the AlgorithmManager loop,
+                // which caused issues with the alignment of data in the statistics calculation phase.
+                SampleBenchmark(time.Date, Algorithm.Benchmark.Evaluate(time).SmartRounding());
+
+                // We need to sample the equity separately from the next if statement method since
+                // we need consistent sampling of the equity curve, rather than potentially having a sparse equity curve in rare cases,
+                // which is used to save on bandwidth costs when uploading interim algorithm results to the cloud.
+                SampleEquity(time, Math.Round(currentPortfolioValue, 4));
+
+                // The benchmark will have an invalid first value since we don't
+                // calculate the percentage change between open and close for the first point.
+                //
+                // The performance series will contain the cumulative percent gain from market open to close
+                // for the first data point. This means that the first point between benchmark and performance
+                // are incompatible with each other, and must be removed for accurate calculations.
+                var portfolioValuePercentChange = _previousDailyPortfolioValue == 0 ? 0 : Math.Round((currentPortfolioValue - _previousDailyPortfolioValue) * 100 / _previousDailyPortfolioValue, 10);
```

Comment:
I think we are missing to tweak the logic to sample the saved portfolio value vs the current, else it seems it will use the current value, which is in the next date already?